include/pstl/internal/glue_algorithm_defs.h

```diff
@@ -0,0 +1,525 @@
+mismatch(ExecutionPolicy&& exec, ForwardIterator1 first1, ForwardIterator1 last1, ForwardIterator2 first2, BinaryPredicate pred);
+
+template< class ExecutionPolicy, class ForwardIterator1, class ForwardIterator2 >
+pstl::internal::enable_if_execution_policy<ExecutionPolicy, std::pair<ForwardIterator1, ForwardIterator2>>
+mismatch(ExecutionPolicy&& exec, ForwardIterator1 first1, ForwardIterator1 last1, ForwardIterator2 first2, ForwardIterator2 last2);
+
+template< class ExecutionPolicy, class ForwardIterator1, class ForwardIterator2 >
+pstl::internal::enable_if_execution_policy<ExecutionPolicy, std::pair<ForwardIterator1, ForwardIterator2>>
+mismatch(ExecutionPolicy&& exec, ForwardIterator1 first1, ForwardIterator1 last1, ForwardIterator2 first2);
+
+// [alg.equal]
+
+template<class ExecutionPolicy, class ForwardIterator1, class ForwardIterator2, class BinaryPredicate>
+pstl::internal::enable_if_execution_policy<ExecutionPolicy, bool>
+equal(ExecutionPolicy&& exec, ForwardIterator1 first1, ForwardIterator1 last1, ForwardIterator2 first2, BinaryPredicate p);
+
+template<class ExecutionPolicy, class ForwardIterator1, class ForwardIterator2>
+pstl::internal::enable_if_execution_policy<ExecutionPolicy, bool>
+equal(ExecutionPolicy&& exec, ForwardIterator1 first1, ForwardIterator1 last1, ForwardIterator2 first2);
+
+template<class ExecutionPolicy, class ForwardIterator1, class ForwardIterator2, class BinaryPredicate>
+pstl::internal::enable_if_execution_policy<ExecutionPolicy, bool>
+equal(ExecutionPolicy&& exec, ForwardIterator1 first1, ForwardIterator1 last1, ForwardIterator2 first2, ForwardIterator2 last2, BinaryPredicate p);
+
+template<class ExecutionPolicy, class ForwardIterator1, class ForwardIterator2>
+pstl::internal::enable_if_execution_policy<ExecutionPolicy, bool>
+equal(ExecutionPolicy&& exec, ForwardIterator1 first1, ForwardIterator1 last1, ForwardIterator2 first2, ForwardIterator2 last2);
+
+// [alg.move]
+template< class ExecutionPolicy, class ForwardIterator1, class ForwardIterator2 >
+pstl::internal::enable_if_execution_policy<ExecutionPolicy, ForwardIterator2>
+move(ExecutionPolicy&& exec, ForwardIterator1 first, ForwardIterator1 last, ForwardIterator2 d_first);
+
+// [partial.sort]
+
+template<class ExecutionPolicy, class RandomAccessIterator, class Compare>
+pstl::internal::enable_if_execution_policy<ExecutionPolicy, void>
+partial_sort(ExecutionPolicy&& exec, RandomAccessIterator first, RandomAccessIterator middle, RandomAccessIterator last, Compare comp);
+
+template<class ExecutionPolicy, class RandomAccessIterator>
+pstl::internal::enable_if_execution_policy<ExecutionPolicy, void>
+partial_sort(ExecutionPolicy&& exec, RandomAccessIterator first, RandomAccessIterator middle, RandomAccessIterator last);
+
+// [partial.sort.copy]
+
+template<class ExecutionPolicy, class _InputIterator, class RandomAccessIterator, class Compare>
```

Comment:
According to N4700 and even N4659 ForwardIterator and RandomAccessIterator are used here.
I believe ForwardIterator should be used instead of _InputIterator as you write below for partial_sort_copy without Compare

Reply:
Thank you for quick fixing